### PR TITLE
Replace beatmap library cache system with a database table using Room API

### LINKED ISSUE #328: Beatmaps may duplicate at certain occasions
This is a bug that has happened to multiple players, yet bizarre as the root cause of the problem is still unknown.

This seems to be a problem at caching operation, as when one of the duplicates is deleted from song selection menu, both beatmaps will be deleted. Additionally, once the player reloads the beatmap cache, the problem is gone, but it can still come back at certain occasions.

Despite being reported multiple times, little is known about this bug. Anyone experiencing the bug or seeing it first-hand can comment more details on this issue.

This is a report that I received from a player. Note that downloading the map from Chimu does not always trigger the problem, otherwise I would have more information to work with. All beatmap cache loading operations navigate towards `LibraryManager.updateLibrary()`.
> All of the maps i have are duplicated and now there is 2 of every map. I started having this problem after i tried the new built in map downloader function, Chimu.moe. After i downloaded a map, it caused all of my maps to be duplicated and if i try to delete one of the copies, both will be deleted.
image

### PR DESCRIPTION
**Pull request by Reco1I**
* Fixes #328

### CODE REVIEW

**Comment by Reco1I** [comment]
PR is ready

File: src/ru/nsu/ccfit/zuev/osu/Config.java [hunk 1]
```diff
@@ -797,4 +802,31 @@ public static boolean isKeepBackgroundAspectRatio() {
...
+        return prefs.getBoolean(key, defaultValue);
+    }
+
+    public static void setString(String key, String value) {
+        prefs.edit().putString(key, value).apply();
+    }
+
+    public static String getString(String key, String defaultValue) {
+        return prefs.getString(key, defaultValue);
+    }
```
[the remark this answers is not in the document]
Comment (Reco1I) [reply src/ru/nsu/ccfit/zuev/osu/Config.java:831]: I forgot, it's something I was thinking about for another PR sorry.

File: src/com/reco1l/osu/Beatmaps.kt [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+     *
+     * [OnConflictStrategy.REPLACE]
+     */
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    fun insert(beatmapInfo: BeatmapInfo): Long
+
+    /**
+     * Delete a beatmap set from the table using it key (MD5 or ID).
+     */
+    @Query("DELETE FROM BeatmapInfo WHERE parentPath = :beatmapSetKey")
```
[the remark this answers is not in the document]
Comment (Reco1I) [reply src/com/reco1l/osu/Beatmaps.kt:297]: Umm how should we resolve this ?

[the remark this answers is not in the document]
Comment (Rian8337) [reply src/com/reco1l/osu/Beatmaps.kt:293]: Solved in https://github.com/osudroid/osu-droid/pull/380/commits/e3ac58093e37ab39db4d632129d428b7347949d2.

**Pull request merged**